pkg/messaging/direct_messaging.go

```diff
@@ -133,6 +142,34 @@ func (d *directMessaging) invokeRemote(ctx context.Context, targetID string, req
 	defer cancel()
 
 	ctx = diag.AppendToOutgoingGRPCContext(ctx, sc)
+
+	metadata := req.Metadata()
```

Comment:
please take this to helper method,  : addForwardedHeaders(...) , keep the main logic to concise.

Reply:
Done